### PR TITLE
Sanitize

### PR DESCRIPTION
**Pull request by williamfgc**
This is an ongoing effort to sanitize piece of the code. It should be regular practice. Related to #1615

### CODE REVIEW

**Comment by williamfgc** [comment]
@NAThompson

File: bindings/C/adios2/c/adios2_c_attribute.cpp [hunk 1]
```diff
@@ -169,7 +169,7 @@ adios2_error adios2_attribute_data(void *data, size_t *size,
                 *size = attributeCpp->m_Elements;
                 char **dataT = reinterpret_cast<char **>(data);
 
-                for (auto e = 0; e < *size; ++e)
+                for (size_t e = 0; e < *size; ++e)
```
Comment (NAThompson) [inline bindings/C/adios2/c/adios2_c_attribute.cpp]: Just a quick comment: `size` is going to become a worse variable name in the future, as C++17 introduces `std::size` and `using std::size` has gotten into various popular C++ libraries. Whether there will actually be a conflict, I don't know.

File: source/adios2/toolkit/format/bp4/BP4Deserializer.tcc [hunk 2]
```diff
@@ -800,7 +800,12 @@ void BP4Deserializer::DefineVariableInEngineIOPerStep(
                             : header.Path + PathSeparator + header.Name;
 
     core::Variable<T> *variable = nullptr;
-    variable = engine.m_IO.InquireVariable<T>(variableName);
+    {
+        // to prevent conflict with DefineVariable
+        std::lock_guard<std::mutex> lock(m_Mutex);
+        variable = engine.m_IO.InquireVariable<T>(variableName);
```
Comment (NAThompson) [inline source/adios2/toolkit/format/bp4/BP4Deserializer.tcc]: Great catch on this one.

File: bindings/C/adios2/c/adios2_c_attribute.cpp (hunk 1 above)
Comment (williamfgc) [reply bindings/C/adios2/c/adios2_c_attribute.cpp]: @NAThompson thanks for pointing this out. We need to be in the lookout for new language features.

**Pull request merged**